build-config/make/crosstool-ng.make

```diff
@@ -43,24 +54,35 @@ crosstool-ng-download: $(CROSSTOOL_NG_DOWNLOAD_STAMP)
 $(CROSSTOOL_NG_DOWNLOAD_STAMP): $(PROJECT_STAMP)
 	$(Q) rm -f $@ && eval $(PROFILE_STAMP)
 	$(Q) echo "==== Getting upstream $(CROSSTOOL_NG_DESC) ===="
-	$(Q) $(SCRIPTDIR)/fetch-package $(DOWNLOADDIR) $(UPSTREAMDIR) \
-		$(CROSSTOOL_NG_TARBALL) $(CROSSTOOL_NG_URLS)
+	$(Q) if [ $(shell awk 'BEGIN{ print "'$(CROSSTOOL_NG_VERSION)'"<="'$(CROSSTOOL_NG_LATEST_RELEASE_VERSION)'" }') -eq 1 ]; then \
+			$(SCRIPTDIR)/fetch-package $(DOWNLOADDIR) $(UPSTREAMDIR) \
+			$(CROSSTOOL_NG_TARBALL) $(CROSSTOOL_NG_URLS); \
+	     fi
 	$(Q) touch $@
 
 SOURCE += $(CROSSTOOL_NG_SOURCE_STAMP)
 crosstool-ng-source: $(CROSSTOOL_NG_SOURCE_STAMP)
 $(CROSSTOOL_NG_SOURCE_STAMP): $(CROSSTOOL_NG_DOWNLOAD_STAMP)
 	$(Q) rm -f $@ && eval $(PROFILE_STAMP)
 	$(Q) echo "==== Extracting upstream $(CROSSTOOL_NG_DESC) ===="
-	$(Q) $(SCRIPTDIR)/extract-package $(CROSSTOOL_NG_BUILD_DIR) $(DOWNLOADDIR)/$(CROSSTOOL_NG_TARBALL)
+	$(Q) if [ $(shell awk 'BEGIN{ print "'$(CROSSTOOL_NG_VERSION)'"<="'$(CROSSTOOL_NG_LATEST_RELEASE_VERSION)'" }') -eq 1 ]; then \
+			$(SCRIPTDIR)/extract-package $(CROSSTOOL_NG_BUILD_DIR) \
+			$(DOWNLOADDIR)/$(CROSSTOOL_NG_TARBALL); \
+	     else \
+			git clone https://github.com/crosstool-ng/crosstool-ng \
+			$(CROSSTOOL_NG_DIR); \
+			cd $(CROSSTOOL_NG_DIR) && git reset --hard $(CROSSTOOL_NG_COMMIT);\
+	     fi	
```

Comment:
OK, sure use `CROSSTOOL_NG_VERSION = custom` as the flag.  Works for me.

I would also put the git repo URL into a variable instead of hard coding it in there.  Something like:
```
CROSSTOOL_NG_GIT_URL ?= clone https://github.com/crosstool-ng/crosstool-ng
```

I would also use `?=` when setting the following:
```
CROSSTOOL_NG_VERSION
CROSSTOOL_NG_COMMIT
CROSSTOOL_NG_GIT_URL
```